swagger.yaml

```diff
@@ -1,75 +1,298 @@
+          in: "query"
+          description: "Number of cards to return"
+          required: false
+          type: integer
+          default: 10
+
       responses:
         200:
-          description: Object containing information
+          description: "Cards successfully retrieved"
+          schema:
+            $ref: "#/definitions/MultiData"
+        400:
+          description: "Invalid request"
+          schema:
+            $ref: "#/definitions/Error"
+        404:
+          description: "Card not found"
+          schema:
+            $ref: "#/definitions/Error"
+        500:
+          description: "Internal server error"
+          schema:
+            $ref: "#/definitions/Error"
+
+    post:
+      tags:
+      - "cards"
+      summary: "Add a new card to the database"
+      description: ""
+      operationId: "addCard"
+      consumes:
+      - "application/json"
+      produces:
+      - "application/json"
+      parameters:
+        - name: "body"
+          in: "body"
+          description: "Card object to be added to the database"
+          required: true
+          schema:
+            $ref: "#/definitions/SingleData"
+
+      responses:
+        201:
+          description: "Card successfully added"
+        400:
+          description: "Invalid syntax in POST request"
           schema:
-            $ref: '#/definitions/Info'
+            $ref: "#/definitions/Error"
+
 
-parameters:
-  pretty:
-    name: pretty
-    in: query
-    type: boolean
-    required: false
-    description: If true, JSON response will be pretty-printed
+  /cards/{id}:
+
+    get:
+      tags:
+      - cards
+      summary: "Retrieve a card by its id"
+      description: ""
+      operationId: "getCardById"
+      consumes:
+      - "application/json"
+      produces:
+      - "application/json"
+      parameters:
+        - name: "id"
+          in: "path"
+          description: "Id of card to be retrieved"
+          required: true
+          type: "integer"
+          format: "int64"
+      responses:
+        200:
+          description: "Card successfully retrieved"
+          schema:
+            $ref: "#/definitions/SingleData"
+        404:
+          description: "Card not found"
+          schema:
+            $ref: "#/definitions/Error"
+
+    put:
+      tags:
+      - cards
+      summary: "Update a card specified by its id"
+      description: ""
+      operationId: "updateCardById"
+      consumes:
+      - "application/json"
+      produces:
+      - "application/json"
+      parameters:
+        - name: "id"
+          in: "path"
+          description: "Id of card to be updated"
+          required: true
+          type: "integer"
+          format: "int64"
+
+        - name: "body"
+          in: "body"
+          description: "Object that will overwrite the card specified by its id"
+          required: true
+          schema:
+            $ref: "#/definitions/SingleData"
+      responses:
+        204:
```

Comment:
Same comment as the POST, it would be good to return the updated object in the response. I also made a mistake with my comment in Jira when recommending you change the status code. I realized a 200 might be the better option because there isn't a good code for "updated".